src/ring_theory/tensor_product.lean

```diff
@@ -483,6 +485,34 @@ instance : comm_ring (A ⊗[R] B) :=
 
 end comm_ring
 
+section left_algebra
+
+variables {R A A' B : Type*} [comm_semiring R] [comm_semiring A] [semiring A'] [semiring B]
+variables [algebra R A] [algebra R A'] [algebra A A'] [is_scalar_tower R A A'] [algebra R B]
+
+instance left_algebra : algebra A (A' ⊗[R] B) :=
+{ commutes' := λ r x,
+  begin
+    apply tensor_product.induction_on x,
+    { simp },
+    { intros a b, dsimp, rw [algebra.commutes, _root_.mul_one, _root_.one_mul] },
+    { intros y y' h h', dsimp at h h' ⊢, simp only [mul_add, add_mul, h, h'] },
+  end,
+  smul_def' := λ r x,
+  begin
+    apply tensor_product.induction_on x,
+    { simp [smul_zero] },
+    { intros a b, dsimp, rw [tensor_product.smul_tmul', algebra.smul_def r a, _root_.one_mul] },
+    { intros, dsimp, simp [smul_add, mul_add, *] },
+  end,
+  .. tensor_product.include_left.to_ring_hom.comp (algebra_map A A'),
+  .. (by apply_instance : module A (A' ⊗[R] B)) }
```

Comment:
The instance above is just a special case of this one, right?

Reply:
Yes. Should I remove that?
This depends on some stuff after that, and I don't think these two instances create diamonds though.

Reply:
I think it would be good to define the one above in terms of this one, simply to reduce the length of the file.